File: #129_Sum_Root_to_Leaf_Numbers/Solution.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

public class Solution {
	int result;
    public int sumNumbers(TreeNode root) {
		if (root == null) return 0;
		result = 0;
		dfs(root, 0);
		return result;
    }

    private void dfs(TreeNode root, int sum) {
    	sum = sum * 10 + root.val;
    	if (root.left == null && root.right == null) {
    		result += sum;
    		return;
    	}
    	if (root.left != null) dfs(root.left, sum);
    	if (root.right != null) dfs(root.right, sum);
    }
}

File: #129_Sum_Root_to_Leaf_Numbers/Solution.py
class TreeNode:
	def __init__(self, x):
		self.val = x
		self.left = None
		self.right = None

class Solution:

	def __init__(self):
		self.sum = 0

	# @param {TreeNode} root
	# @return {integer}
	def sumNumbers(self, root):
		self.sum = 0
		self.dfs(root, 0)
		return self.sum

	def dfs(self, root, pathSum):
		if not root:
			return
		if not root.left and not root.right:
			self.sum += pathSum * 10 + root.val
			return
		self.dfs(root.left, pathSum * 10 + root.val)
		self.dfs(root.right, pathSum * 10 + root.val)

sol = Solution()
root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
print sol.sumNumbers(root)